### PR TITLE
Add support for passing title and alt text to `useDamFileUpload`

### PR DESCRIPTION
**Pull request by VP-DS**
## Description

This enables passing title and alt text as custom properties onto file objects. The license information is also moved into this custom metadata to enable the upload of an array of files with different license information using the `uploadFiles` function.

## Screenshots/screencasts

https://github.com/user-attachments/assets/75f5e139-e39b-49a0-a3c3-2fa136c01805

In the demo I added the data to the file like this for testing:
Screenshot 2024-12-06 at 14 07 53

### CODE REVIEW

File: demo/admin/src/dam/getRandomUnsplashImage.ts [hunk 1]
```diff
@@ -1,8 +1,20 @@
...
+class ExtendedFile extends File {
+    license?: GQLLicenseInput;
+    title?: string;
+    altText?: string;
+
+    constructor(blob: BlobPart[], fileName: string, options: FilePropertyBag) {
+        super(blob, fileName, options);
+    }
+}
+
```
Comment (VP-DS) [inline demo/admin/src/dam/getRandomUnsplashImage.ts:17]: should this be moved into the package and be imported by the project?

btw. should the demo set randomized license, title, altText

File: demo/admin/src/dam/getRandomUnsplashImage.ts [hunk 2]
```diff
@@ -23,7 +35,7 @@ function extractFileNameFromUrl(url: string): string {
 }
 
 export async function getRandomUnsplashImage(): Promise<UnsplashImage> {
-    const imageUrl = "https://source.unsplash.com/all/";
+    const imageUrl = "https://picsum.photos/1920/1080"; // https://picsum.photos/ uses unsplashed images
```
Comment (VP-DS) [inline demo/admin/src/dam/getRandomUnsplashImage.ts:38]: off-topic, but found this which uses unsplashed images too but works fine, seems like a good replacement

**Comment by VP-DS** [comment]
In the demo I added the data to the file like this:
Screenshot 2024-12-06 at 14 07 53

File: demo/admin/src/dam/getRandomUnsplashImage.ts (hunk 2 above)
Comment (VP-DS) [reply demo/admin/src/dam/getRandomUnsplashImage.ts]: In the demo I added the data to the file like this:
Screenshot 2024-12-06 at 14 07 53

**Review by thomasdax98 — APPROVED** [review]

File: demo/admin/src/dam/getRandomUnsplashImage.ts (hunk 2 above)
Comment (thomasdax98) [reply demo/admin/src/dam/getRandomUnsplashImage.ts]: Nice find, we should switch that

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**